src/index.ts

```diff
@@ -22,6 +22,18 @@ import BN from 'bn.js'
 import { osmosis, thorchain } from './amino'
 import * as codecs from './proto'
 
+type AgnosticStdTx = Omit<amino.StdTx, 'msg'> &
+  (
+    | {
+        readonly msg?: readonly amino.AminoMsg[]
+        msgs?: never;
+      }
+    | {
+        readonly msgs?: readonly amino.AminoMsg[]
+        msg: never;
```

Comment:
`msg` optional never (though, see above about different way to define this type that makes more sense to me)